src/main/java/com/rtb/projectmanagementtool/project/ProjectData.java

```diff
@@ -140,56 +126,120 @@ public void setDescription(String description) {
   }
 
   /**
-   * Adds an admin user to project
+   * Add a regular user to the project
+   *
+   * @param userId id of the user to add
+   */
+  public void addRegularUser(long userId) {
+    addUser(userId, false);
+  }
+
+  /**
+   * Add an admin user to the project
    *
    * @param userId id of the user to add
    */
   public void addAdminUser(long userId) {
-    // remove user if they're already in project
-    if (this.userIds.keySet().contains(userId)) {
-      removeUser(userId);
-    }
-    this.userIds.put(userId, User.ADMIN);
+    addUser(userId, true);
   }
 
   /**
-   * Adds a regular user to the project
+   * Helper function for addRegularUser() and addAdminUser() to add any user to the project. Can
+   * only add regular/admin users
    *
    * @param userId id of the user to add
+   * @param isAdmin is the user an admin?
    */
-  public void addRegularUser(long userId) {
+  private void addUser(long userId, boolean isAdmin) {
     // remove user if they're already in project
-    if (this.userIds.keySet().contains(userId)) {
+    if (hasUser(userId)) {
       removeUser(userId);
     }
-    this.userIds.put(userId, User.REGULAR);
+
+    if (isAdmin) {
+      this.userIds.put(userId, UserProjectRole.ADMIN);
+    } else {
+      this.userIds.put(userId, UserProjectRole.REGULAR);
+    }
   }
 
   /**
    * Removes a user from the project
    *
    * @param userId id of the user to remove
+   * @return true if operation is successful
    */
-  public void removeUser(long userId) {
-    this.userIds.remove(userId);
+  public boolean removeUser(long userId) {
+    // don't remove creator
+    if (userType(userId) != UserProjectRole.CREATOR.name()) {
```

Comment:
If the CREATOR is special enough and doesn't act as the two other users (cannot be added or remove), consider moving the CREATOR into it's own createroUserID field.